crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -289,7 +287,7 @@ impl CachingNativePriceEstimator {
         results
     }
 
-    pub fn replace_high_priority(&self, tokens: HashSet<H160>) {
+    pub fn replace_high_priority(&self, tokens: IndexSet<H160>) {
```

Comment:
For curiosity, why are you changing from `HashSet` to `IndexSet`?

Reply:
But `HashSet` wouldn't have worked because it doesn't keep track of insertion order which is important for this algorithm to work. With a `HashSet` we could only say whether a token was prioritized or not which is meaningless given that all tokens get assigned a priority. With an `IndexSet` we not only know whether an item is part of the set but also when it was inserted (i.e. the priority).

Reply:
I understand now! thanks a lot @MartinquaXD for the explanation! highly appreciated!